packages/client/src/generation/generateClient.ts

```diff
@@ -135,6 +135,25 @@ export async function buildClient({
     fileMap['edge.d.ts'] = await TS(edgeTsClient, true)
   }
 
+  if (generator?.previewFeatures.includes('denoDeploy') && !!globalThis.Deno) {
+    // we create a client that is fit for edge runtimes
+    const denoTsClient = new TSClient({
+      ...tsClientOptions,
+      dataProxy: true, // edge only works w/ data proxy
+      runtimeName: 'index.d.ts',
+      runtimeDir: '../' + runtimeDirs.edge,
+      denoDeploy: true,
+    })
+
+    fileMap['deno/edge.js'] = await JS(denoTsClient, true)
+    fileMap['deno/index.d.ts'] = await TS(denoTsClient)
+    fileMap['deno/edge.ts'] = `import 'https://deno.land/std@0.152.0/dotenv/load.ts'
+import './polyfill.js'
+// @deno-types="./index.d.ts"
+export * from './edge.js'`
+    fileMap['deno/polyfill.js'] = 'globalThis.process = { env: Deno.env.toObject() }; globalThis.global = globalThis'
```

Comment:
I'm wondering if this would also work?
```suggestion
    fileMap['deno/index.js'] = await JS(denoTsClient, true)
    fileMap['deno/edge.d.ts'] = await TS(denoTsClient)
    fileMap['deno/edge.js'] = `import 'https://deno.land/std@0.152.0/dotenv/load.ts'
import './polyfill.js'
export * from './index.js'`
    fileMap['deno/polyfill.js'] = 'globalThis.process = { env: Deno.env.toObject() }; globalThis.global = globalThis'
```

Reply:
Renaming `deno/edge.js` -> `deno/index.js` and `deno/index.d.ts` -> `deno/edge.d.ts` is fine, but renaming `deno/edge.ts` to `deno/edge.js` is problematic. Deno doesn't automatically pick up types from `deno/edge.d.ts` for `deno/edge.js`. If we do the above change, the users don't get the type completion anymore. The entrypoint for the users needs to be `.ts` to make types available

Reply:
Thanks, this sentence is a good summary for me.

> Deno doesn't automatically pick up types from deno/edge.d.ts